.github/workflows/ci.yml

```diff
@@ -288,3 +288,10 @@ jobs:
 
             echo $git_status
             test -z "$git_status"
+
+  ensure-lockfile-uptodate:
+    name: Ensure that `Cargo.lock` is up-to-date
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: cargo metadata --frozen --format-version=1 > /dev/null
```

Comment:
@thomaseizinger was it a deliberate choice to use `--frozen` instead of `--locked` here?

CI is failing as with `--frozen` it is not able to update its crates.io index.

https://github.com/libp2p/rust-libp2p/actions/runs/4386547439/jobs/7680746875

Unfortunately, given that this step is not yet tracked as a required step, and given that I did not pay proper attention, I merged this already.

Reply:
Interesting. Yes it was deliberate because `--locked` didn't work on my machine for `cargo metadata`. I'll investigate.